csharp/ql/test/utils/model-generator/theorems/CaptureTheoremsForFreeSummaryModels.expected

```diff
@@ -0,0 +1,14 @@
+| Summaries;CollectionTheorems1<>;false;Add;(T);;Argument[0];Argument[this].SyntheticField[ArgT].Element;taint;generated |
+| Summaries;CollectionTheorems1<>;false;First;();;Argument[this].SyntheticField[ArgT].Element;ReturnValue;taint;generated |
+| Summaries;Theorems1<>;false;Apply;(System.Func<T,System.Int32>);;Argument[this].SyntheticField[ArgT];Argument[0].Parameter[0];taint;generated |
+| Summaries;Theorems1<>;false;Get;();;Argument[this].SyntheticField[ArgT];ReturnValue;taint;generated |
+| Summaries;Theorems1<>;false;Get;(System.Object);;Argument[this].SyntheticField[ArgT];ReturnValue;taint;generated |
+| Summaries;Theorems1<>;false;Id;(T);;Argument[0];ReturnValue;taint;generated |
```

Comment:
I don't think this is the summary we want. I'd expect a `T -> T` method on a `Type<T>` to be a simultaneous getter and setter, i.e. something like "replace" or "addAndReturnExisting". The Java `V Map::put(K, V)` method e.g. returns an existing `MapValue` while also adding a `MapKey` and a `MapValue`.

Reply:
Makes sense. In general this must mean that our assumptions is that `T -> S` is a setter no matter the type of `S`: If `S = T` then your reasoning holds and if S != T (eg. S=unit or S is just something unrelated) then the only thing we can do with the value of type `T` is to set it (at least as long as there are no type constraints on `T`). The dual reasoning can be performed for signatures like `S -> T` (these must then be getter like).